QModManager/HarmonyPatches/EnableConsoleSetting.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace QModManager.HarmonyPatches.EnableConsoleSetting
+{
+    using Harmony;
+    using QModManager.Utility;
+    using System.Collections.Generic;
+    using System.Reflection;
+    using System.Reflection.Emit;
+    using UnityEngine;
+
+    [HarmonyPatch(typeof(DevConsole), nameof(DevConsole.Awake))]
+    internal static class DevConsole_Awake_Patch
+    {
+        // This patch toggles the console based on the mod option
+
+        [HarmonyPostfix]
+        internal static void Postfix()
+        {
+            DevConsole.disableConsole = !Config.EnableConsole;
+        }
+    }
+
+    [HarmonyPatch(typeof(GraphicsDebugGUI), nameof(GraphicsDebugGUI.OnGUI))]
+    internal static class GraphicsDebugGUI_OnGUI_Patch
+    {
+        // This patch prevents the F3 debug menu from toggling the console
+
+        [HarmonyTranspiler]
+        internal static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
+        {
+            foreach (CodeInstruction instruction in instructions)
+            {
+                if (instruction.opcode == OpCodes.Stsfld && (FieldInfo)instruction.operand == AccessTools.Field(typeof(DevConsole), nameof(DevConsole.disableConsole)))
```

Comment:
Create a temporary variable above the `foreach` so that this only needs to be calculated once
`AccessTools.Field(typeof(DevConsole), nameof(DevConsole.disableConsole)`

Also, you don't need to cast `instruction.operand` for this evaluation.

Reply:
![image](https://user-images.githubusercontent.com/32238504/72435779-952c5c80-37a7-11ea-9b15-c9e113aed2c0.png)